src/common/Formatter.cc

```diff
@@ -34,8 +34,8 @@
 #include <boost/format.hpp>
 
 
-static char tolower_underscore(const char b) {
-  return ' ' == b ? '_' : std::tolower(b);
+static char underscore(const char b) {
```

Comment:
could move this into a private const method of `XMLFormatter`, so we don't need to duplicate the same pattern everywhere. like
```c++
  char to_lower_underscore(char c) const {
    if (c == ' ') {
      if (m_underscored) {
        return '_';
      }
    } else if (m_lowercased) {
      return tolower(c);
    }
    return c;
  }
```
and do the transform in a single pass
```c++
    std::transform(e.begin(), e.end(), e.begin(), [this](unsigned char c) {
        return this->to_lower_underscore(c);
      });
```